typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/endItemDecoratorStrategy.ts

```diff
@@ -0,0 +1,119 @@
+import { getDecoratorProps } from "../../../helpers/node";
+import CanvasItemDecoratorStrategy from "./canvasItemDecoratorStrategy";
+import { buildItemParameterBindings, InputOutputBindings } from "./helpers/presentation";
+
+// UI positioning constants in the output XML file.
+const xBasePosition = 160;
+const yBasePosition = 100;
+const offSet = 20;
+
+export default class EndItemDecoratorStrategy implements CanvasItemDecoratorStrategy {
+	/**
+	 * Return XML tag for the end workflow item.
+	 * 
+	 * @returns XML tag name.
+	 */
+	getCanvasType(): string {
+		return "end";
+	}
+
+	/**
+	 * Return the workflow item type supported by this decorator.
+	 * 
+	 * @returns type of the workflow element.
+	 */
+	getDecoratorType(): WorkflowItemType {
+		return WorkflowItemType.End;
+	}
+
+	/**
+	 * Register the canvas item arguments. For the default error handler only "endMode" and "exception" are supported.
+	 * 
+	 * @param itemInfo item info for that properties should be fetched.
+	 * @param decoratorNode decorator node handle.
+	 */
+	registerItemArguments(itemInfo: WorkflowItemDescriptor, decoratorNode: Decorator): void {
+		const decoratorProperties: [string, any][] = getDecoratorProps(decoratorNode);
+		if (!decoratorProperties?.length) {
+			return;
+		}
+		decoratorProperties.forEach((propTuple) => {
+			const [propName, propValue] = propTuple;
+			switch (propName) {
+				case "endMode": {
+					itemInfo.canvasItemPolymorphicBag.endMode = propValue;
+					break;
+				}
+				case "exception": {
+					itemInfo.canvasItemPolymorphicBag.exception = propValue;
+					break;
+				}
+				case "businessStatus": {
+					itemInfo.canvasItemPolymorphicBag.businessStatus = propValue;
+					break;
+				}
+				default: {
+					throw new Error(`Item attribute '${propName}' is not supported for ${this.getDecoratorType()} item`);
+				}
+			}
+		});
+	}
+
+	/**
+	 * There is no need to print the source file for the workflow item.
+	 */
+	printSourceFile(methodNode: MethodDeclaration, sourceFile: SourceFile): string {
+		return "";
+	}
+
+	/**
+	 * Prints out the end item.
+	 *
+	 * @param itemInfo The item to print.
+	 * @param pos The position of the item in the workflow.
+	 *
+	 * @returns The string representation of the item.
+	 */
+	printItem(itemInfo: WorkflowItemDescriptor, pos: number): string {
+		const stringBuilder = new StringBuilderClass("", "");
+
+		const endMode = itemInfo?.canvasItemPolymorphicBag?.endMode;
+		const exceptionVariable = itemInfo?.canvasItemPolymorphicBag?.exception;
+		const businessStatus = itemInfo?.canvasItemPolymorphicBag?.businessStatus;
+		const hasException = (exceptionVariable !== null && exceptionVariable !== undefined && exceptionVariable);
+		const hasBusinessStatus = (businessStatus !== null && businessStatus !== undefined && businessStatus);
+
+		stringBuilder.append(`<workflow-item name="item${pos}" type="end" end-mode="${endMode}" `);
```

Comment:
Maybe same as other PR, you can work on the formatting here and do it at once for readability